### DISCUSSION TITLE
Parser for emojis?

### DISCUSSION THREAD

**Discussion by nank1ro** [petitparser/dart-petitparser#176] [Q&A]
Hi, thank you so much for this amazing package.
I've built [notescalculator.com](https://notescalculator.com) using it and it helped a lot.

May I know If there is a way to accept emojis in a parser?

I have this parser now
```dart
  late final Parser<String> variable = [
    char(' ').star(),
    letter(),
    pattern('a-zA-Z_0-9 ').star().optional()
  ].toSequenceParser().flatten().map((value) => value.trim());
```

But I would like it to accept something like `👍 hello` or `hel👍lo`

**Comment by renggli** [comment]
Thank you for the feedback, and for sharing your amazing application. Works very nicely on the web!

Unfortunately detecting [emoji characters](https://unicode.org/Public/emoji/15.1/) in Unicode is not trivial. The easiest way to go about it is probably to use the package [emoji_regex](https://pub.dev/packages/emoji_regex). Its complicated regular expression can be used with the `PatternParser` which seems to work well in my test below:

```dart
import 'package:emoji_regex/emoji_regex.dart';
import 'package:petitparser/petitparser.dart';

final emoji =
    PatternParser(emojiRegex(), 'Emoji expected').map((each) => each[0]);

void main() {
  const input = '😊, 🐪, 🇯🇵, 🙋🏿, 👨‍👩‍👧‍👦';
  print(emoji.allMatches(input).join('\n'));
}
```

Possibly it would be more efficient to implement the regular expression using PetitParser patterns, but to get started this is the easiest solution I can think of.

**Comment by nank1ro** [comment]
Thank you so much for the help! It worked perfectly fine 🎉
